### PR TITLE
Hannah Cameron - Hotel - Octos

### PR DESCRIPTION
**Pull request by hannahlcameron**
# Hotel
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe a design decision you had to make when working on this project. What options were you considering? What helped you make your final decision? | A design decision I made was to make Block a subclass of Reservation. I was also considering having Block be its own independent class. I ended up making this decision because it seemed like Reservations and Blocks shared a lot of common attributes and behaviors. Looking back, I wish I had better thought through this process from the beginning and made a superclass for both reservation and block. I ended up running into issues where my current design became overly complicated when attempting to make a reservation within a block.
Describe a concept that you gained more clarity on as you worked on this assignment. | I specifically chose to use a subclass within this project because I wanted to have another chance at using inheritance. 
Describe a _nominal test_ that you wrote for this assignment. | A nominal test I wrote as a part of this assignment is a part of the HotelManager spec file: it "returns the correct amount of room ids". This test is nominal because it takes in expected input and tests for expected output.
Describe an _edge case test_ that you wrote for this assignment. |  An edge case is in reservation spec: it "raises an error if provided an invalid date range". This tests for if someone enters unexpected input - a start date that is after an end date. 
How do you feel you did in writing pseudocode first, then writing the tests and then the code? | I feel really good about the process of pseudocode, then tests and then real code. I find the TDD method really helpful to follow - it helps me stay on track working on one small part and then another.

### CODE REVIEW

File: lib/hotel_manager.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      return room_list
+    end
+
+    def get_reservation_id
+
+      if @reservations == []
+        next_reservation_id = 1
+        # puts "The next reservation will be #{next_reservation_id}"
+        return next_reservation_id
+      else
```
Comment (kariabancroft) [inline lib/hotel_manager.rb:27]: This else logic will actually still work if the array is empty so you don't need the conditional

**Comment by kariabancroft** [comment]
# Hotel
## What We're Looking For

Feature | Feedback
---     | ---
**Design** |
Demonstrated classes having a single responsibility | Yes. Nice job making an intentional choice to practice inheritance. 
Demonstrated loose coupling |Yes
Methods demonstrate a good use of encapsulation, inputs and outputs | Yes
Wave 1 requirements |Yes
Wave 2 requirements |Yes
Wave 3 requirements | Incomplete

Nice job!
